### PR TITLE
Sandboxing for expression evaluation

### PR DESCRIPTION
**Pull request by stephenfreund**
Function calls in expressions are wrapped in a test for whether they appear in a white list.  The whitelist enumerates either full modules or specific functions using regular expressions in a text file.  There is a new option to use a different file.  Right now it includes the modules necessary for all of our tests (I think), plus all the built ins except `eval` and `exec`.

### CODE REVIEW

File: src/chatdbg/chatdbg_pdb.py [hunk 1]
```diff
@@ -13,8 +13,9 @@
 
 import IPython
 
-import chatdbg.custom_pdb as pdb
```
Comment (nicovank) [inline src/chatdbg/chatdbg_pdb.py:16]: Is this a voluntary change? Emery just changed this, just making sure it's not inadvertent.

AFAIK we shouldn't name files/modules the same name as Python builtins, it causes bugs.

File: src/chatdbg/util/module_whitelist.txt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+builtins.slice
+builtins.sorted
+builtins.staticmethod
+builtins.str
+builtins.sum
+builtins.super
+builtins.tuple
+builtins.type
+builtins.vars
+builtins.zip
```
Comment (nicovank) [inline src/chatdbg/util/module_whitelist.txt:72]: Any reason not to `builtins.*` as above?

Comment (stephenfreund) [reply src/chatdbg/util/module_whitelist.txt:72]: eval and exec...  Fixing the broken test, etc. now.

Comment (nicovank) [reply src/chatdbg/util/module_whitelist.txt:72]: Makes sense.

File: src/chatdbg/chatdbg_pdb.py (hunk 1 above)
Comment (stephenfreund) [reply src/chatdbg/chatdbg_pdb.py:16]: Yes, in this case, we actually wanted the **builtin** pdb module.  I had imported it to access the `pdb.Pdb` class.  Emery's change broke the code but it must have gone unnoticed.

Comment (nicovank) [reply src/chatdbg/chatdbg_pdb.py:16]: Okay makes sense. It was probably only tested with LLDB and not PDB. Thanks for fixing 🙏

Comment (stephenfreund) [reply src/chatdbg/chatdbg_pdb.py:16]: Also: I don't believe the two files left in the `custom_pdb` directory are used anywhere.  The only references I can find are in cover-up tests...   I won't do it in this PR, but I think we can just delete that directory.

Comment (nicovank) [reply src/chatdbg/chatdbg_pdb.py:16]: ACK, I’m a fan of deleting code, if it’s unused go for it.

**Review by nicovank — APPROVED** [review]

**Pull request merged**